### PR TITLE
Reversed Rainbow

### PR DESCRIPTION
**Pull request by TreyRuffy**
This PR changes the way you can create rainbows. If you use the `^` symbol (subject to change) you can create a rainbow which is an exact reversal of a rainbow with or without phase.

The syntax is: `<rainbow:^2>` where `^` reverses the rainbow and `2` changes the phase of the rainbow.

### CODE REVIEW

**Comment by kezz** [comment]
Check the style issues, there's a gradle task you can run to validate it locally. Other than that, I like the idea of being able to reverse a rainbow.

**Comment by TreyRuffy** [comment]
> Check the style issues, there's a gradle task you can run to validate it locally. Other than that, I like the idea of being able to reverse a rainbow.

Sounds good, just pushed the fix

**Comment by thiccaxe** [comment]
Maybe look into `!` as opposed to `^`, it has more meaning because of its usage as not

**Comment by TreyRuffy** [comment]
> Maybe look into `!` as opposed to `^`, it has more meaning because of its usage as not

I didn't even think of that. I'll leave the discussion open for a bit before changing it, but I think it's a much better symbol.

File: src/main/java/net/kyori/adventure/text/minimessage/transformation/inbuild/RainbowTransformation.java [hunk 1]
```diff
@@ -79,10 +80,17 @@ public void load(final String name, final List<TagPart> args) {
...
 
     if (args.size() == 1) {
-      try {
-        this.phase = Integer.parseInt(args.get(0).value());
-      } catch (final NumberFormatException ex) {
-        throw new ParsingException("Expected phase, got " + args.get(0), this.argTokenArray());
+      String value = args.get(0).value();
+      if (args.get(0).value().startsWith("^")) {
+        this.reversed = true;
+        value = value.replace("^", "");
```
Comment (kashike) [inline src/main/java/net/kyori/adventure/text/minimessage/transformation/inbuild/RainbowTransformation.java:86]: Reference a `static final String REVERSE` constant in this class instead, please.

**Review by kashike — CHANGES_REQUESTED** [review]

**Comment by MiniDigger** [comment]
I like this, could you create a PR for documenting this on the docs repo before we merge? https://github.com/KyoriPowered/adventure-docs

**Review by kezz — APPROVED** [review]
`!` is much better!

**Comment by MiniDigger** [comment]
sorry in the delay handling this, I moved the REVERSE constant into the token class, but else looks good, thanks!

**Pull request merged**